lib/middlewares/src/metering.rs

```diff
@@ -36,53 +41,63 @@ pub struct FunctionMetering<F: Fn(&Operator) -> u64 + Copy + Clone + Send + Sync
     /// Function that maps each operator to a cost in "points".
     cost_function: F,
 
+    /// The global index in the current module for the error code.
+    error_code_index: GlobalIndex,
+
     /// The global index in the current module for remaining points.
     remaining_points_index: GlobalIndex,
 
     /// Accumulated cost of the current basic block.
     accumulated_cost: u64,
 }
 
+#[derive(Debug, PartialEq)]
+pub enum MeteringError {
+    NoError,
+    OutOfGas,
```

Comment:
The term "gas" is not used elsewhere in the implementation as far as I know. I'd go with something like

```suggestion
    OutOfPoints,
```